fastlane_bot/bot.py

```diff
@@ -312,55 +272,36 @@ def _run(
         Parameters
         ----------
         flashloan_tokens: List[str]
-            The tokens to flashloan.
+            The flashloan tokens
         CCm: CPCContainer
-            The container.
+            The complete market data container
         arb_mode: str
-            The arbitrage mode.
+            The arbitrage mode
         randomizer: int
-            randomizer (int): The number of arb opportunities to randomly pick from, sorted by expected profit.
-        data_validator: bool
-            If extra data validation should be performed
+            The number of top arbitrage opportunities to randomly choose from
         logging_path: str
-            the logging path (default: None)
-        replay_mode: bool
-            whether to run in replay mode (default: False)
+            The logging path
         replay_from_block: int
-            the block number to start replaying from (default: None)
-
+            The block number to start replaying from
         """
-        arbitrage = self._find_arbitrage(flashloan_tokens=flashloan_tokens, CCm=CCm, arb_mode=arb_mode, randomizer=randomizer)
-        finder, r = [arbitrage[key] for key in ["finder", "r"]]
 
-        if r is None or len(r) == 0:
-            self.ConfigObj.logger.info("[bot._run] No eligible arb opportunities.")
+        if flashloan_tokens is None:
+            flashloan_tokens = self.RUN_FLASHLOAN_TOKENS
+        if CCm is None:
+            CCm = self.get_curves()
+
+        arb_finder = self.get_arb_finder(arb_mode, flashloan_tokens, CCm)
+        arb_opps = arb_finder.find_arb_opps()
+
+        if len(arb_opps) == 0:
+            self.ConfigObj.logger.info("[bot.run] No eligible arb opportunities.")
             return
 
-        self.ConfigObj.logger.info(
-            f"[bot._run] Found {len(r)} eligible arb opportunities."
-        )
-        r = self.randomize(arb_opps=r, randomizer=randomizer)
+        self.ConfigObj.logger.info(f"[bot.run] Found {len(arb_opps)} eligible arb opportunities.")
 
-        if data_validator:
-            r = self.validate_optimizer_trades(arb_opp=r, arb_finder=finder)
-            if r is None:
-                self.ConfigObj.logger.warning(
-                    "[bot._run] Math validation eliminated arb opportunity, restarting."
-                )
-                return
-            if replay_mode:
-                pass
-            elif self.validate_pool_data(arb_opp=r):
-                self.ConfigObj.logger.debug(
-                    "[bot._run] All data checks passed! Pools in sync!"
-                )
-            else:
-                self.ConfigObj.logger.warning(
-                    "[bot._run] Data validation failed. Updating pools and restarting."
-                )
-                return
+        arb_opp = rand_item(list_of_items=arb_opps, num_of_items=randomizer)
```

Comment:
this is consistent with what we are doing now, but arguably randomisation should happen at the node level because they have the complete picture; we should just return everything we find